chroma_core/plugins/block_devices.py

```diff
@@ -12,23 +12,51 @@
 log.setLevel(DEBUG)
 
 
-def _fetch_aggregator():
+def _fetch_aggregator(timeout):
     import requests
 
-    resp = requests.get(settings.DEVICE_AGGREGATOR_URL)
-    payload = resp.text
-
-    return json.loads(payload)
-
-
-def get_devices(fqdn):
-    try:
-        _data = _fetch_aggregator()
-        return _data[fqdn]
-    except Exception as e:
-        log.error(
-            "iml-device-aggregator is not providing expected data, ensure "
-            "iml-device-scanner package is installed and relevant "
-            "services are running on storage servers (%s)" % e
-        )
-        return {}
+    summary = 0.0
+    timeout = timeout if timeout >= 0 else 0
```

Comment:
If someone passes in a negative number, is it better to change it to 0 than raise an error?

Reply:
I think so. At least, from a point of backward compatibility.
If you wish, you can suggest a proper exception name.